src/Cli/Microsoft.TemplateEngine.Cli/Commands/InteractiveMode/InteractiveMode.cs

```diff
@@ -0,0 +1,253 @@
+                    parser.Invoke(commandLineCall);
+                }
+            }
+            else
+            {
+                await next(context).ConfigureAwait(false);
+            }
+        }
+
+        internal static InstantiateCommand? GetInstantiateCommand(Command command)
+        {
+            NewCommand? newCommand = command as NewCommand;
+            InstantiateCommand? instantiateCommand;
+            if (newCommand is null)
+            {
+                instantiateCommand = command as InstantiateCommand;
+            }
+            else
+            {
+                instantiateCommand = newCommand.Subcommands.First(command => command.Name == "create") as InstantiateCommand;
+            }
+
+            return instantiateCommand;
+        }
+
+        internal static InstantiateCommandArgs? GetArgs(ParseResult parseResult)
+        {
+            NewCommand? newCommand = parseResult.CommandResult.Command as NewCommand;
+            InstantiateCommandArgs? knownArgs = null;
+            if (newCommand is null)
+            {
+                var instantiateCommand = parseResult.CommandResult.Command as InstantiateCommand;
+                if (instantiateCommand is not null)
+                {
+                    knownArgs = new InstantiateCommandArgs(instantiateCommand, parseResult);
+                }
+            }
+            else
+            {
+                var newCommandArgs = new NewCommandArgs(newCommand, parseResult);
+                knownArgs = InstantiateCommandArgs.FromNewCommandArgs(newCommandArgs);
+            }
+
+            return knownArgs;
+        }
+
+        internal static bool ShouldEnterInteractiveMode(ParseResult parsedArgs)
+        {
+            // TODO: find a better way to query for the option
+            // TODO: Better handling of this option when executing
+            bool interactiveOptionValue = false;
+            if (NewCommand.InteractiveTemplateOption is not null)
+            {
+                interactiveOptionValue = parsedArgs.GetValue(NewCommand.InteractiveTemplateOption);
+            }
+
+            var envVariable = Environment.GetEnvironmentVariable("DOTNETNEWINTERACTIVE");
+            bool envVariableValue = Convert.ToBoolean(envVariable);
+
+            if ((interactiveOptionValue || envVariableValue) && DoesSupportInteractive())
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        internal static bool DoesSupportInteractive()
+        {
+            // hard coded for now
+            // Check if current environment supports what we want to do (If we need any fancy stuff)
+            return true;
+        }
+
+        internal static TemplateCommand? GetTemplate(
+            InstantiateCommandArgs args,
+            IEngineEnvironmentSettings environmentSettings,
+            IEnumerable<TemplateGroup> templateGroups,
+            TemplatePackageManager templatePackageManager)
```

Comment:
if calling method is moved to `InstantiateCommand` it should be possible to reuse those methods.